src/Bridges/Constraint/set_map.jl

```diff
@@ -5,91 +5,98 @@
-functions must be implemented:
-
- * [`MOI.Bridges.inverse_map_set`](@ref)
- * [`MOI.Bridges.inverse_map_function`](@ref)
- * [`MOI.Bridges.adjoint_map_function`](@ref)
- * [`MOI.Bridges.inverse_adjoint_map_function`](@ref)
+    abstract type MultiSetMapBridge{T,S1,G} <: AbstractBridge end
+
+Same as `SetMapBridge` but the output constraint type does not only depend on
+the input constraint type.  When subtyping `MultiSetMapBridge`,
+`added_constraint_types` and `supports` should additionally be implemented by
+the bridge.  For instance, if a bridge `BridgeType` may create either a
+constraint of type `F2`-in-`S2` or `F3`-in-`S3`, these methods may be
+implemented as follows:
+```julia
+function MOI.Bridges.added_constraint_types(
+    ::Type{<:BridgeType{T,F2,F3}},
+) where {T,F2,F3}
+    return Tuple{Type,Type}[(F2, S2), (F3, S3)]
+end
 
-See the docstrings of each function to see which feature would be missing if it
-was not implemented for a given bridge.
+function MOI.supports(
+    model::MOI.ModelLike,
+    attr::Union{MOI.ConstraintPrimalStart,MOI.ConstraintDualStart},
+    ::Type{<:BridgeType{T,F2,F3}},
+) where {T,F2,F3}
+    return MOI.supports(model, attr, MOI.ConstraintIndex{F2,S2}) ||
+        MOI.supports(model, attr, MOI.ConstraintIndex{F3,S3})
+end
+```
 """
-abstract type SetMapBridge{T,S2,S1,F,G} <: AbstractBridge end
+abstract type MultiSetMapBridge{T,S1,G} <: AbstractBridge end
 
 function bridge_constraint(
-    BT::Type{<:SetMapBridge{T,S2,S1,F,G}},
+    BT::Type{<:MultiSetMapBridge{T,S1,G}},
     model::MOI.ModelLike,
     func::G,
     set::S1,
-) where {T,S2,S1,F,G}
+) where {T,S1,G}
     mapped_func = MOI.Bridges.map_function(BT, func)
     mapped_set = MOI.Bridges.map_set(BT, set)
     constraint = MOI.add_constraint(model, mapped_func, mapped_set)
     return BT(constraint)
 end
 
 function MOI.supports_constraint(
-    ::Type{<:SetMapBridge{T,S2,S1}},
+    ::Type{<:MultiSetMapBridge{T,S1}},
     ::Type{<:MOI.AbstractScalarFunction},
     ::Type{S1},
-) where {T,S2,S1<:MOI.AbstractScalarSet}
+) where {T,S1<:MOI.AbstractScalarSet}
     return true
 end
 
 function MOI.supports_constraint(
-    ::Type{<:SetMapBridge{T,S2,S1}},
+    ::Type{<:MultiSetMapBridge{T,S1}},
     ::Type{<:MOI.AbstractVectorFunction},
     ::Type{S1},
-) where {T,S2,S1<:MOI.AbstractVectorSet}
+) where {T,S1<:MOI.AbstractVectorSet}
     return true
 end
 
-function MOI.Bridges.added_constrained_variable_types(::Type{<:SetMapBridge})
+function MOI.Bridges.added_constrained_variable_types(
+    ::Type{<:MultiSetMapBridge},
+)
     return Tuple{Type}[]
 end
 
-function MOI.Bridges.added_constraint_types(
-    ::Type{<:SetMapBridge{T,S2,S1,F}},
-) where {T,S2,S1,F}
-    return Tuple{Type,Type}[(F, S2)]
-end
-
 # Attributes, Bridge acting as a model
 
-function MOI.get(
-    ::SetMapBridge{T,S2,S1,F},
-    ::MOI.NumberOfConstraints{F,S2},
-)::Int64 where {T,S2,S1,F}
-    return 1
+_get(::MOI.ConstraintIndex, ::MOI.NumberOfConstraints) = 0
```

Comment:
I've found that these "helper" functions just make things more difficult in the long run. Let's just explicitly write out the implementation for both constraint types.